spec/controllers/reviews_controller_spec.rb

```diff
@@ -0,0 +1,162 @@
+require 'rails_helper'
+
+RSpec.describe ReviewsController, type: :controller do
```

Comment:
Great tests! Excellent for the context of the drill - but for testing the Controller I'd focus on the simpler flows for the controller itself, and validate input sanity at the ActiveRecord level (specially where the validations happen on the ActiveRecord level) , like you did with the shoulda matchers.
I would spend time on changing it now.